kouta-backend/src/main/scala/fi/oph/kouta/repository/helpers.scala

```diff
@@ -15,25 +15,29 @@ import scala.util.{Failure, Success, Try}
 
 trait SQLHelpers extends KoutaJsonFormats with Logging {
 
-  def createOidInParams(x: Seq[Oid]) = x.find(!_.isValid) match {
+  def createOidInParams(x: Seq[Oid]): String = x.find(!_.isValid) match {
     case None if x.isEmpty => s"''"
     case Some(i) => throw new IllegalArgumentException(s"$i ei ole validi oid.")
     case None => x.map(s => s"'$s'").mkString(",")
   }
 
-  def createUUIDInParams(x: Seq[UUID]) = if( x.isEmpty) s"''" else x.map(s => s"'${s.toString}'").mkString(",")
+  def createUUIDInParams(x: Seq[UUID]): String = if( x.isEmpty) s"''" else x.map(s => s"'${s.toString}'").mkString(",")
 
-  def createRangeInParams(x: Seq[Ajanjakso]) = if(x.isEmpty) s"''" else x.map(s => s"${toTsrangeString(s)}").mkString(",")
+  def createRangeInParams(x: Seq[Ajanjakso]): String = if(x.isEmpty) s"''" else x.map(s => s"${toTsrangeString(s)}").mkString(",")
 
-  def formatTimestampParam(value: Option[LocalDateTime]) = value.map(ISO_LOCAL_DATE_TIME_FORMATTER.format).getOrElse(null)
+  def formatTimestampParam(value: Option[LocalDateTime]): String = value.map(ISO_LOCAL_DATE_TIME_FORMATTER.format).orNull
 
-  def toJsonParam(value: AnyRef) = Option(toJson(value)) match {
+  def toJsonParam(value: AnyRef): String = Option(toJson(value)) match {
     case Some(s) if !s.isEmpty & !"{}".equals(s) => s
     case _ => null
   }
 
   def createKoulutustyypitInParams(x: Seq[Koulutustyyppi]): String = if (x.isEmpty) "''" else x.map(tyyppi => s"'${tyyppi.name}'").mkString(",")
 
+  def createArkistoidutFilter(myosArkistoidut: Boolean): String = {
+    if (myosArkistoidut) "" else s"and tila <> '$Arkistoitu'"
+  }
```

Comment:
Se ehkä hieman huonontaa niiden itse SQL:ien luettavuutta DAO:issa, kun tuo and on täällä eikä se näy mitenkään siellä DAO:n puolella. Siksi vaihtaisin tämän nimeksi esim. `def andMaybeNotArkistoitu` tai jotain vastaavaa.

Reply:
Joo tämä oli inhottava kun "and" piti ehdollisuuden takia tähän laittaa eikä voinut jättää sql stringiin. Harkitsin jopa koko sql:n monistamista että tuon olisi välttänyt mutta ei sekään oikein hyvältä tuntunut. Tosi hyvä idea tuo että nimeää funktion käyttöpaikan dsl:n mukaisesti!